castle/utils.py

```diff
@@ -1,20 +1,31 @@
 import copy
+from datetime import datetime
 
 def clone(dict_object):
     return copy.deepcopy(dict_object)
 
 def deep_merge(base, extra):
-    """Deeply two dictionaries, overriding existing keys in the base.
+    """
+    Deeply merge two dictionaries, overriding existing keys in the base.
+
     :param base: The base dictionary which will be merged into.
     :param extra: The dictionary to merge into the base. Keys from this
         dictionary will take precedence.
     """
-    for key in extra:
+    if extra is None:
+        return
+
+    for key, value in extra.items():
```

Comment:
because you're supporting python 2 and 3, I would add a dependency on `six` and do

```python
from six import iteritems
...
for key, value in iteritems(extra)
```

Reply:
with sdks I prefer to limit dependencies to total minimum (zero preferred). we can discuss performance improvements in next PR